backend/xivdpscalc/character/__init__.py

```diff
@@ -57,7 +57,7 @@ def calc_piety(self):
         return 200 + self.stats[Stats.PIE].get_multiplier()
 
     # todo: break this up neatly
-    def calc_damage(self, potency, comp, is_dot=False, crit_rate=None, dh_rate=None):  #pylint: disable=too-many-arguments, too-many-locals
+    def calc_damage(self, potency: float, comp: Comp, is_dot: bool = False, crit_rate: float = None, dh_rate: float = None):  #pylint: disable=too-many-arguments, too-many-locals
```

Comment:
Small nit here:

`crit_rate` and `dh_rate` are annotated as a `float`, but we default these values to `None`. Might be better to use `typing.Optional` and annotate as `Optional[float]` – the reason being is that `None` is not a float, but an *optional* float can be None. For side reference, the python docs on [Optional](https://docs.python.org/3/library/typing.html#typing.Optional) support that `Optional[foo]` is analogous  to `Union[foo, None]`.